### ISSUE TITLE
[fr] read only

### ISSUE THREAD

**Issue by throwaway242685** [FossifyOrg/Notes#69]
### Checklist

- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Notes/issues) or [closed](https://github.com/FossifyOrg/Notes/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Notes/discussions) or [closed](https://github.com/FossifyOrg/Notes/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Notes/blob/master/CONTRIBUTING.md).
- [ ] I optionally [donated](https://fossify.org/donate) to support the Fossify mission.

### Feature description

hi, it would be great if Notes had an option/button to set notes as "read only'.

### Why do you want this feature?

I personally save text to use/copy later on these notes, and I don't want to accidentally overwrite the text on these notes when I tap on them.

this would be different than locking the note in the sense that you would see the text being displayed, but you wouldn't be able to edit it.

**Comment by vbernier02** [comment]
Hello,

I am a third-year computer science student at the University. As part of a course of free software development, i would like to work on this feature for my first contribution.

**Comment by throwaway242685** [comment]
yeah, sure, go ahead.

I'm sure both Aga-C and naveensingh will be okay with that.

what do you think about implementing a new "lock" button that would put the note to "read only" and viceversa?

something like this:

![Screenshot_2025-01-28_05-18-31_1.png](https://github.com/user-attachments/assets/61f1e7d2-cc8e-4443-824c-005dd5d5e350)

**Comment by Aga-C** [comment]
@throwaway242685 This icon is currently used for locked notes. We can't use this icon in another context because it will be misleading  for users.

**Comment by throwaway242685** [comment]
I meant at the top, as a very small icon, next to the others.

any other icon design with this function will be okay, I guess.

**Comment by vbernier02** [comment]
Hi, for the feature i'm thinking of keeping the project's approach and adding it to the options menu, similar to the "lock note" feature. 
I was thinking like, pressing the option will lock the writing on the note and you will have to return to the option to unlock it.
I'll keep the idea in mind for an icon.

**Comment by naveensingh** [comment]
Actually, let's wait a bit until we have a better idea on how to approach this. I do not like the menu button idea.

**Comment by throwaway242685** [comment]
well, if it eventually comes down to using a button:

![screenshot_20250128-080134.png](https://github.com/user-attachments/assets/e4032506-0a6e-4388-95e5-a3a6a3e205aa)

I was thinking that it should be placed at the top left corner, separated from the rest of the icons.

that way is less prone to miss-click, or things like that.

and, it's more visible :-)

**Comment by naveensingh** [comment]
> I was thinking that it should be placed at the top left corner, separated from the rest of the icons.

That's not the right place for it _and_ it's reserved for the back button, see https://github.com/FossifyOrg/Notes/issues/27

> I do not like the menu button idea.

@vbernier02

I am not happy with it but if you are up for it, let's do it. You can check [Material Notes](https://f-droid.org/en/packages/com.maelchiotti.localmaterialnotes
) for reference. The final icons depend on the text, here's some examples:
 - [Enter edit mode](https://fonts.google.com/icons?selected=Material+Symbols+Rounded:edit:FILL@0;wght@400;GRAD@0;opsz@24&icon.query=edit&icon.size=24&icon.color=%23000000&icon.style=Rounded&icon.platform=web)
 - [Enter preview mode](https://fonts.google.com/icons?selected=Material+Symbols+Rounded:visibility:FILL@0;wght@400;GRAD@0;opsz@24&icon.query=view&icon.size=24&icon.color=%23000000&icon.style=Rounded&icon.platform=web)
  - [Enter read-only mode](https://fonts.google.com/icons?selected=Material+Symbols+Rounded:edit_off:FILL@0;wght@400;GRAD@0;opsz@24&icon.query=edit&icon.size=24&icon.color=%23000000&icon.style=Rounded&icon.platform=web)

(I like `Enter preview mode` because then we won't need another button for https://github.com/FossifyOrg/Notes/issues/22)

The `Create a new note` action button should probably be moved back into the overflow menu because we are already [breaking the rule](https://m3.material.io/components/top-app-bar/guidelines#15a358a9-3305-486f-9ce9-d0c939e02d2a) with 5 existing buttons:

<img src="https://github.com/user-attachments/assets/9c9d2b6a-ee41-4015-98d0-b2df739e86e6" width=264 />

**Comment by vbernier02** [comment]
>I am not happy with it but if you are up for it, let's do it.

Thanks, I'll work on it. I'll try to do it with the icons you suggested for now. If it seems better, we can put the feature in the overflow menu.

**Comment by vbernier02** [comment]
Hi, I've been working on the feature for a while now but I have a problem that I can't fix for a few weeks. When I edit a note and test my read only implementation, the modification added to the note are not save.

https://github.com/user-attachments/assets/45ed1285-1b7a-49c2-877a-f7fb499394ca

I don't know if it's because I'm touching the TextFragment or something else. If anyone has a clue for my problem, I'll take it.
The fork if you want to check : https://github.com/vbernier02/Notes

Thanks

**Comment by throwaway242685** [comment]
@vbernier02 I wish I could help you, but I'm not a programmer myself :(

hopefully someone else will help you with that.

**Comment by naveensingh** [comment]
@throwaway242685 

Would you want to be able to select/copy text while in read only mode?

**Comment by throwaway242685** [comment]
> Would you want to be able to select/copy text while in read only mode?

yes, that's actually what I had on mind :) that you could select/copy text with the keyboard, but the keys that you press in the keyboard won't edit the note at all.

> Should this affect checklists as well or just text notes?

I personally don't use checklists, but I guess it should be the same.

**Comment by cyberbeat** [comment]
What about this? I need this feature, for that the keyboard doesn't popup, when I only want to read it with full screen height.
